source/core.py

```diff
@@ -69,6 +69,53 @@ def __bool__(self):
 _shuttingDownFlagLock = threading.Lock()
 
 
+def _showAddonsWarnings() -> None:
+	addonFailureMessages: list[str] = []
+	failedUpdates = addonHandler._failedPendingInstalls.intersection(addonHandler._failedPendingRemovals)
+	failedInstalls = addonHandler._failedPendingInstalls - failedUpdates
+	failedRemovals = addonHandler._failedPendingRemovals - failedUpdates
+	if failedUpdates:
+		addonFailureMessages.append(
+			ngettext(
+				# Translators: Shown when one or more add-ons failed to update.
+				"The following add-on failed to update: {}",

```

Comment:
Add end of sentence period. It's better when reading the whole dialog (NVDA+B) and various such messages are concatenated.

```suggestion
				"The following add-on failed to update: {}.",
```